### PR TITLE
Add Alpine 3.20 to OS tests matrix

### LINKED ISSUE #3503: Add Alpine 3.20 to OS testing matrix
### Is your feature request related to a problem? Please describe.

Alpine 3.20 isn't currently tested in the OS testing matrix.

### Describe the solution you would like

Add it to the matrix.

### Describe alternatives you've considered

_No response_

### Additional context

I've had problems getting SSH logins to work back in the day, that's why it hasn't been added until now. Probably just needs the right config settings.

### PR DESCRIPTION
**Pull request by ncopa**
## Description

Add Alpine Linux 3.20 to the OS test matrix, including arm64.

Fixes #3503

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by twz123 — COMMENTED** [review]
Nice!

**Comment by twz123** [comment]
https://github.com/k0sproject/k0s/actions/runs/9957443949
https://github.com/k0sproject/k0s/actions/runs/9957453920

File: hack/ostests/README.md [hunk 1]
```diff
@@ -69,7 +69,8 @@ terraform apply
 ### `os`: Operating system stack
 
 * `al2023`: Amazon Linux 2023
-* `alpine_3_17`: Alpine Linux 3.17
```
[the remark this answers is not in the document]
Comment (ncopa) [reply hack/ostests/README.md:72]: not yet. Its EOL in november.

File: hack/ostests/modules/os/os_alpine_3_20_ready.sh
[the remark this answers is not in the document]
Comment (ncopa) [reply hack/ostests/modules/os/os_alpine_3_20_ready.sh:1]: lets drop the suffix for now, and reintroduce if ever needed.

File: .github/workflows/ostests-e2e.yaml [hunk 2]
```diff
@@ -113,7 +113,7 @@ jobs:
           jq --version
 
           mkdir -p "$(dirname -- "$TF_VAR_k0sctl_executable_path")"
-          curl --proto '=https' --tlsv1.2 --retry 5 --retry-all-errors -sSLfo "$TF_VAR_k0sctl_executable_path" "https://github.com/k0sproject/k0sctl/releases/download/${K0SCTL_VERSION}/k0sctl-linux-x64"
+          curl --proto '=https' --tlsv1.2 --retry 5 --retry-all-errors -sSLfo "$TF_VAR_k0sctl_executable_path" "https://github.com/k0sproject/k0sctl/releases/download/${K0SCTL_VERSION}/k0sctl-linux-${OSTESTS_ARCH/amd64/x64}"
```
Comment (twz123) [inline .github/workflows/ostests-e2e.yaml:116]: Hmm, k0sctl is executed on the runner, and the runner arch is always amd64 (runner arch != cluster arch) ... :thinking:

/cc #4772

**Review by twz123 — APPROVED** [review]

**Pull request merged**